core/src/main/java/nl/nn/adapterframework/http/rest/ApiListener.java

```diff
@@ -91,6 +91,8 @@ public enum HttpMethod {
 	private @Getter String requiredClaims=null;
 	private @Getter String exactMatchClaims=null;
 	private @Getter String roleClaim;
+
+	private @Getter String principalClaim = "sub";
```

Comment:
I suggest renaming this to `principalNameClaim`, because it is about getting the name of the principal rather than the actual ID of the principal.